inst/shiny/modules/tlg/tlg_option_table.R

```diff
@@ -0,0 +1,158 @@
+#' Function generating an input widget for TLG option.
+#' @param id      id of the input widget
+#' @param opt_def definition of the option, as specified in the `yaml` file
+#' @param data    data object used for parsing labels, strings, infering placeholder values or
+#'                choices etc.
+tlg_option_table_ui <- function(id, opt_def, data) {
+  ns <- NS(id)
+
+  label <- if (is.null(opt_def$label)) id else opt_def$label
+
+  actionButton(ns("open_table"), label = label)
+}
+
+#' Function generating an input widget server for TLG option.
+#' @param id      id of the input widget
+#' @param opt_def definition of the option, as specified in the `yaml` file
+#' @param data    data object used for parsing labels, strings, infering placeholder values or
+#'                choices etc.
+#' @returns a reactive with the input value
+tlg_option_table_server <- function(id, opt_def, data) {
+  moduleServer(id, function(input, output, session) {
+    #' Generates default table based on provided default option definition
+    default_table <- lapply(opt_def$default_rows, \(x) {
+      as.list(x) %>%
+        setNames(names(opt_def$cols)) %>%
+        as_tibble() %>%
+        mutate(across(everything(), as.character))
+    }) %>%
+      bind_rows() %>%
+      mutate(across(everything(), ~ ifelse(. == "$NA", NA, .)))
+
+    #' Output table with values that is returned by the module
+    output_table <- reactiveVal(default_table)
+
+    #' Table for holding edits provided by the user, before confirming.
+    edits_table <- reactiveVal(default_table)
+
+    #' For controling reactable re-rendering
+    refresh_reactable <- reactiveVal(0)
+
+    #' Shows the editing table
+    observeEvent(input$open_table, {
+      .tlg_option_table_popup()
+    })
+
+    #' Renders the editing table
+    output$table <- renderReactable({
+      #' Create edit widgets using `reactable.extras` based on provided definitions
+      edit_widgets <- imap(opt_def$cols, \(def, colname) {
+        colDef(
+          cell = switch(
+            def$type,
+            text = text_extra(id = session$ns(colname)),
+            select = dropdown_extra(
+              id = session$ns(colname),
+              choices = {
+                if (isTRUE(def$choices == ".colnames")) {
+                  names(data())
+                } else if (length(def$choices) == 1 && grepl("^\\$", def$choices)) {
+                  unique(data()[, sub("^\\$", "", def$choices)])
+                } else {
+                  def$choices
+                }
+              },
+              class = "dropdown-extra"
+            ),
+            stop("Unsupported extra")
```

Comment:
Maybe a more informative error message here? something like:

`stop("Unsupported type: ", def$type, " for column: ", colname)`